### PR TITLE
Clear screen on invalid move

### PR DESCRIPTION
**Pull request by RabeaGleissner**
Hi @felipesere and @jsuchy,
This is the PR for the story "clear screen on invalid move". 

I also updated the user prompt which asks if the user wants to play a HvH or HvC game because Jim said the previous version made him think too much :wink: 

I also had some maven compile errors, so I had to update my POM to include the Java version.

### CODE REVIEW

File: src/main/java/de/rabea/ui/UserInterface.java [hunk 1]
```diff
@@ -12,7 +12,7 @@ public UserInterface(Console console) {
     private String askUserForPosition = "Please select a position for your mark.";
     private String wantToPlayAgain = "Do you want to play again? y/n";
     private String greeting = "Welcome to Tic Tac Toe.\n ";
-    private String gameOptions = "Please enter 1 if you want to play against the computer and 2 if you want to play against another human player.\n";
+    private String gameOptions = "Please choose the game mode. \n 1 - Human vs Computer\n 2 - Human vs Human";
```
Comment (jsuchy) [inline src/main/java/de/rabea/ui/UserInterface.java:15]: Separation and delineation makes things bettah :)

File: src/main/java/de/rabea/ui/UserInterface.java [hunk 2]
```diff
@@ -20,14 +20,21 @@ public UserInterface(Console console) {
...
+    private String clearScreen= "\033[H\033[2J";
 
 
     public void displayBoard(Board board) {
+        clearTheScreen();
         BoardPainter boardPainter = new BoardPainter(board);
         console.print(boardPainter.drawBoard());
     }
 
+    private void clearTheScreen() {
```
Comment (jsuchy) [inline src/main/java/de/rabea/ui/UserInterface.java:32]: asymmetry between this method and the previous. Either it's <verb>Board or <verb>TheBoard.

File: src/main/java/de/rabea/ui/UserInterface.java [hunk 3]
```diff
@@ -88,6 +95,7 @@ public boolean playAgain() {
 
     private boolean userReplayChoice(String userChoice) {
         if (inputFormatter.formatForReplayOption(userChoice) == Replay.YES) {
+            clearTheScreen();
```
Comment (jsuchy) [inline src/main/java/de/rabea/ui/UserInterface.java:98]: Should we have side-effects in query methods?

File: src/main/java/de/rabea/ui/UserInterface.java [hunk 4]
```diff
@@ -96,7 +104,9 @@ private boolean userReplayChoice(String userChoice) {
         }
     }
 
-    public void positionUnavailableWarning() {
+    public void positionUnavailableWarning(Board board) {
+        clearTheScreen();
+        displayBoard(board);
```
Comment (jsuchy) [inline src/main/java/de/rabea/ui/UserInterface.java:109]: do you ever displayBoard without clearTheScreen? Might be duplication here.

File: src/test/java/de/rabea/ui/UserInterfaceTest.java [hunk 5]
```diff
@@ -102,7 +102,7 @@ public void returnsHvHGameModeWhenUserEnters1() {
     public void asksTheUserAgainForGameModeIfBadInputIsEntered() {
         fakeConsole.userInput("hello", "1");
         userInterface.chooseGameMode();
-        assertEquals("Please enter 1 if you want to play against the computer and 2 if you want to play against another human player.\n", fakeConsole.messagePrinted());
```
Comment (jsuchy) [inline src/test/java/de/rabea/ui/UserInterfaceTest.java:105]: I want one and only one test to fail if we have to change a detail of the user interface. Do we need both of these tests? Or can we DRY it up so only one test fails the next time we make a UI change?

**Comment by RabeaGleissner** [comment]
Thanks for your comments @jsuchy ! All makes sense.
I implemented the changes directly on the master branch in this commit https://github.com/RabeaGleissner/java-tic-tac-toe/commit/95c632bc4adf6b040d26150d087e732a1251d956

**Pull request merged**